vspec/__init__.py

```diff
@@ -900,13 +930,20 @@ def check_data_type_references_across_trees(
     nonexistant_types = []
     for _, _, node in RenderTree(signal_root):
         # signals with user-defined data types
+        append_node = False
         if node.is_signal() and node.datatype is None:
-            if not node.does_attribute_exist(
+            if data_type_root is None:
+                # No user-defined types exist
```

Comment:
@kkoppolu1 - I updated this PR, rebasing it on your changes and changes unit behavior somewhat according to the discussion in #217.

I then found what seems to be a bug, if you do not specify any datatype file but use something that is not a primitive type (like `int7`) you will get an exception as `data_type_root` is None. In this PR I added a check here, but we could alternatively make sure that data_type_root never is None.

Reply:
Indeed, this method should not be called if there is no data type tree specified on the command line. I'll add in the check.

Reply:
I have fixed this in https://github.com/COVESA/vss-tools/pull/232
The change is identical. So, you can either revert this particular change. 
Or, I can wait for this to merge and re-base.